text/0123-improve-span-status-api.md

```diff
@@ -95,55 +95,32 @@ translation effort from the native representation and capture at least the key
 numeric/string data developers are likely to search for or relate to. I suggest
 status is this set of information:
 
-1. StatusType - The name of the type of error such as "HTTP", "gRPC",
-"LanguageException", "HRESULT", "POSIX", or "ErrorMessage". The list is
-end-user-extensible but common status type names should be standardized.
-(Perhaps there is already some standardization we could borrow?)
-2. StatusData - A discriminated union of:
-   - An integer, a string, or a tuple of integer and string. These options can
-be used for:
-     - Enumerated status codes: For example an http status code could be
-represented as 404, "Not Found", or both. In the case of common status codes
-OpenTelemetry SDK or backend could optionally assist in filling out the
-remainder of a partially specified enumeration value. For enumerations that
-aren't well-known the community of users is responsible for determining any
-conventions.
-     - Free-form error messages
-   - Exception object - whatever the SDK language's default exception datatype
-is, if it has one.
-   - void
-3. SuccessHint - An optional boolean that represents the span author's best
-guess whether this status represents a successful or failed operation, however
-they choose to define those terms. For well-known status types I'd suggest the
-hint be ignored but for user-defined status types this is likely the only clue
-whether the span should be surfaced in a UI as being abnormal or failed.
+1. Domain - The name of the domain the status data applies to such as "HTTP",
+   "gRPC", "HRESULT", "POSIX". The list is end-user-extensible but common status
+   type names should be standardized.
+   (Perhaps there is already some standardization we could borrow?)
+2. Code - An integer status code. Can be combined with an status message. Either
+   a code or message are required.
+3. Message - A string status message. Can be combined with a status code. Either
+   a code or message are required.
 
 Although UI creators are free experiment with how the data is presented I
 expect most presentations would either be the StatusData alone, or the
 StatusData qualified with the StatusType and some separator character. For
-example StatusData alone might create names like "FileNotFoundException",
-"503", "E_FAIL (0x80004005)", "SyntaxError on line 405: Did you forget a
-semicolon?", and "BadQuery".
-
-Exceptions could have progressive level of detail drilling into messages,
-stack traces, inner exceptions, links to source, etc if the exporter serialized
-sufficient data but how and whether that occurs is out-of-scope in this design. 
+example StatusData alone might create names like "503", "E_FAIL (0x80004005)",
+"Status Code 12: Unimplemented".
```

Comment:
This needs to be updated since `StatusData` was renamed to `Code` and `StatusType` to `Domain` above and `Message` was introduced as a separate parameter.